src/PublishPort.jsx

```diff
@@ -47,18 +93,31 @@
                             min={1}
                             max={65535}
                             value={item.hostPort || ''}
-                            onChange={(_event, value) => onChange(idx, 'hostPort', value)} />
+                            validated={validationFailed?.hostPort ? "error" : "default"}
+                            onChange={(_event, value) => {
+                                utils.validationClear(validationFailed, "hostPort", onValidationChange);
+                                utils.validationDebounce(() => onValidationChange(validatePublishPort(value, "hostPort", validationFailed)));
+                                onChange(idx, 'hostPort', value);
+                            }} />
+                <FormHelper helperTextInvalid={validationFailed?.hostPort} />
             </FormGroup>
             <FormGroup className="pf-m-2-col-on-md"
+                        id={id + "-container-port-group"}
                         label={_("Container port")}
                         fieldId={id + "-container-port"} isRequired>
                 <TextInput id={id + "-container-port"}
                             type='number'
                             step={1}
                             min={1}
                             max={65535}
+                            validated={validationFailed?.containerPort ? "error" : "default"}
                             value={item.containerPort || ''}
-                            onChange={(_event, value) => onChange(idx, 'containerPort', value)} />
+                            onChange={(_event, value) => {
+                                utils.validationClear(validationFailed, "containerPort", onValidationChange);
+                                utils.validationDebounce(() => onValidationChange(validatePublishPort(value, "containerPort", validationFailed)));
+                                onChange(idx, 'containerPort', value);
+                            }} />
+                <FormHelper helperTextInvalid={validationFailed?.containerPort} />
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231016-130710-ce1ba8a5-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)